s3/src/test/scala/akka/stream/alpakka/s3/scaladsl/S3SourceSpec.scala

```diff
@@ -17,14 +18,28 @@ class S3SourceSpec extends S3WireMockBase with S3ClientIntegrationSpec {
     mockDownload()
 
     //#download
-    val s3Source: Source[ByteString, NotUsed] = s3Client.download("testBucket", "testKey")
+    val s3Source: Source[ByteString, NotUsed] = s3Client.download(bucket, bucketKey)
     //#download
 
     val result: Future[String] = s3Source.map(_.utf8String).runWith(Sink.head)
 
     result.futureValue shouldBe body
   }
 
+  it should "download a range of file's bytes from S3 if bytes range given" in {
+
+    mockRangedDownload()
+
+    //#rangedDownload
+    val s3Source: Source[ByteString, NotUsed] =
+      s3Client.download(bucket, bucketKeyForRanged, ByteRange(bytesRangeStart, bytesRangeEnd))
+    //#rangedDownload
+
+    val result: Future[Array[Byte]] = s3Source.map(_.toArray).runWith(Sink.head)
```

Comment:
Taking into account that `Range` header operates on bytes level, I believe, It's better not to couple to `String`s and concrete strings encodings (UTF-8 in this case, but it could be easily be another encoding).
In short, I think, the test should verify that given byte range of data is returned regardless of what this bytes represent.